proof/invariant-abstract/IpcCancel_AI.thy

```diff
@@ -365,10 +367,10 @@ lemma blocked_cancel_ipc_invs:
    apply (frule st_tcb_at_state_refs_ofD)
    apply (subgoal_tac "epptr \<notin> set (remove1 t queue)")
     apply (case_tac ep, simp_all add: valid_ep_def)[1]
-     apply (auto elim!: delta_sym_refs pred_tcb_weaken_strongerE
-                 simp: obj_at_def is_ep_def2 idle_not_queued refs_in_tcb_bound_refs
-                 dest: idle_no_refs
-                 split: if_split_asm)[2]
+     apply (timeit \<open>auto elim!: delta_sym_refs pred_tcb_weaken_strongerE
+                         simp: obj_at_def is_ep_def2 idle_not_queued refs_in_tcb_bound_refs
+                         dest: idle_no_refs
+                         split: if_split_asm\<close>)[2] (* slow: ~100s *)
```

Comment:
The last two times I've left the timeit in there for these very long commands, the reasoning being that when someone is doing changes in future, there's immediate feedback on whether the performance changed significantly (i.e. you get that info even if you already built all the proofs, don't have to re-trigger it and lose progress). You're right, the benefit is a bit marginal. What do you think, keep or remove?

Reply:
Hmm, I do like that benefit now that you've pointed it out. I reckon might as well keep it, it shouldn't be too confusing if we stumble across it in the future.

That does remind me though, I wish that the timing panel in Isabelle had some way of showing which entry corresponds to the command the cursor is currently on. If it had that then you could easily get the performance info without the `timeit`.

Reply:
`timeit` will work even when defer proofs is on (although it will only show output at the end of the proof block), but the timing panel will only show timing for the entire proof block/lemma, so I will leave it for now

Reply:
My vote is also for leaving the `timeit` in. Longer-term it would probably be nice to speed this proof up a bit, but that's out of scope for this PR.